main.go

```diff
@@ -2,54 +2,79 @@ package main
 
 import (
 	"encoding/json"
+	"fmt"
+	"io"
 	"log"
 	"os"
 	"os/exec"
 	"path/filepath"
+	"reflect"
+	"regexp"
 	"strconv"
 	"strings"
 
 	"golang.org/x/exp/slices"
 
-	dbTypes "github.com/aquasecurity/trivy-db/pkg/types"
 	k8sReport "github.com/aquasecurity/trivy/pkg/k8s/report"
 	"github.com/aquasecurity/trivy/pkg/report"
 	"github.com/aquasecurity/trivy/pkg/types"
 )
 
 var (
-	tempJsonFileName = "csv-report-temp.json"
-	templateFileName = "csv.tpl"
+	tempJsonFileName     = "csv-report-temp.json"
+	tempTemplateFileName = "csv-report-template-temp.tpl"
+	templateFileName     = "csv.tpl"
+	availableFieldsNames = []string{
+		"Target",
+		"Vulnerability Class",
+		"Target Type",
+		"Vulnerability ID",
+		"Severity",
+		"PackageName",
+		"Installed Version",
+		"Fixed Version",
+		"Title",
+		"Description",
+		"Resolution",
+		"Reference",
+		"Additional Reference",
+		"CVSS V3 Score",
+		"CVSS V3 Vector",
+	}
+	availableFieldsMap = map[string]bool{}
+	availableFlags     = []string{"--csv-result", "--csv-delimiter", "--csv-include", "--csv-exclude"}
+	delimiter          = ","
 )
 
 func init() {
+	if delimiter = getFlagValue("--csv-delimiter"); delimiter == "" {
+		delimiter = ","
+	}
+	initializeAvailableFields()
+	availableFieldKeys := make([]string, 0, len(availableFieldsNames))
+	for _, field := range availableFieldsNames {
+		if availableFieldsMap[strings.ToLower(field)] {
+			availableFieldKeys = append(availableFieldKeys, "\""+field+"\"")
+		}
+	}
 	var CustomTemplateFuncMap = map[string]interface{}{
-		"escapeCsv": func(input string) string {
-			quoted := strconv.Quote(input)
-			return strings.ReplaceAll(quoted, "\\\"", "\"\"")
-		},
-		"escapeString": func(input string) dbTypes.SourceID {
-			return dbTypes.SourceID(input)
-		},
-		"nvdV3Score": func(input dbTypes.VendorCVSS) float64 {
-			return input["nvd"].V3Score
-		},
-		"rhV3Score": func(input dbTypes.VendorCVSS) float64 {
-			return input["redhat"].V3Score
-		},
-		"nvdV3Vector": func(input dbTypes.VendorCVSS) string {
-			return input["nvd"].V3Vector
-		},
-		"rhV3Vector": func(input dbTypes.VendorCVSS) string {
-			return input["redhat"].V3Vector
+		"getAvailableFields": func() string {
+			return strings.Join(availableFieldKeys, ",")
 		},
+		"getVulnerabilitiesTable":  getVulnerabilitiesTable,
+		"getMisconfigurationTable": getMisconfigurationTable,
 	}
 	report.CustomTemplateFuncMap = CustomTemplateFuncMap
 }
+
 func main() {
-	trivyCommand := os.Args[1 : len(os.Args)-1]
-	outputFileName := os.Args[len(os.Args)-1]
-	tempFileName := filepath.Join(os.TempDir(), tempJsonFileName)
+	trivyCommand := excludePluginFlags(os.Args, availableFlags)[1:]
```

Comment:
now the plugin and trivy arguments can be obtained using the package `github.com/afdesk/trivy-go-plugin/pkg/common`